src/core/src/op/scatter_elements_update.cpp

```diff
@@ -78,256 +71,127 @@ shared_ptr<Node> op::v12::ScatterElementsUpdate::clone_with_new_inputs(const Out
+                                 const op::v12::ScatterElementsUpdate::Reduction reduction_type,
+                                 const bool use_init_value) {
+            reference::scatter_elem_update(data,
+                                           indices.data<IT>(),
+                                           updates,
+                                           normalized_axis,
+                                           output,
+                                           data_shape,
+                                           indices_shape,
+                                           reduction_type,
+                                           use_init_value);
+            return true;
+        }
+    };
+};
 }  // namespace scatter_elements_update
 
-bool op::v3::ScatterElementsUpdate::evaluate_scatter_elements_update(const HostTensorVector& outputs,
-                                                                     const HostTensorVector& inputs) const {
-    const auto normalized_axis = get_normalized_axis(inputs);
-
-    return scatter_elements_update::evaluate_scatter_elements_update(inputs[0],
-                                                                     inputs[1],
-                                                                     inputs[2],
-                                                                     inputs[3],
-                                                                     outputs[0],
-                                                                     normalized_axis);
-}
-
-bool op::v12::ScatterElementsUpdate::evaluate_scatter_elements_update(const HostTensorVector& outputs,
-                                                                      const HostTensorVector& inputs) const {
-    const auto normalized_axis = get_normalized_axis(inputs);
-
-    return scatter_elements_update::evaluate_scatter_elements_update(inputs[0],
-                                                                     inputs[1],
-                                                                     inputs[2],
-                                                                     inputs[3],
-                                                                     outputs[0],
-                                                                     normalized_axis,
-                                                                     m_reduction,
-                                                                     m_use_init_val);
-}
-
-bool op::v3::ScatterElementsUpdate::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool op::v3::ScatterElementsUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v3_ScatterElementsUpdate_evaluate);
-    return evaluate_scatter_elements_update(outputs, inputs);
-}
-
-bool op::v12::ScatterElementsUpdate::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+    OPENVINO_ASSERT(inputs.size() == 4);
```

Comment:
Similar comment as for v12

Why not keep `evaluate_scatter_elements_update` and duplicate code?

Make as local free function instead member function